scripts/provisioning-terraform.sh

```diff
@@ -168,6 +168,17 @@ TERRAFORM_TFVARS_PATH="${TERRAFORM_ENVIRONMENT_DIR}/terraform.tfvars"
 
 # "${HOME}"/.config/gcloud/application_default_credentials.json is a well-known location for application-default credentials
 GOOGLE_APPLICATION_CREDENTIALS_PATH="/root/.config/gcloud/application_default_credentials.json"
+RUNTIME_SCRIPT_FOLDER="$(mktemp -d)"
+
+cat <<EOF >"${RUNTIME_SCRIPT_FOLDER}/run.sh"
```

Comment:
Why do you need to generate this script dynamically? Because of the environment variables? If so, I'd suggest creating a function that takes the needed parameters instead.

Reply:
So the need here is to check if the storage bucket exists and create one if it's not there.
Essentially I didn't generate a script but runs it directly in the container, and the command output does not output to the console, hence the check would always goes to "Create bucket" route.

So I instead generate a script here instead.

Reply:
I'm not sure I get why can't you move these to a script still...

Reply:
Created two functions:
1. Ensure storage bucket exists, if not, create one.
2. Delete storage bucket.